x-pack/legacy/plugins/infra/public/pages/logs/analysis/page_results_content.tsx

```diff
@@ -4,13 +4,187 @@
 
-  return <div>Results</div>;
+  const {
+    timeRange,
+    setTimeRange,
+    autoRefreshEnabled,
+    setAutoRefresh,
+  } = useLogAnalysisResultsUrlState();
+
+  const [refreshInterval, setRefreshInterval] = useState(300000);
+
+  const setTimeRangeToNow = useCallback(() => {
+    const range = timeRange.endTime - timeRange.startTime;
+    const nowInMs = moment()
+      .utc()
+      .valueOf();
+    setTimeRange({
+      startTime: nowInMs - range,
+      endTime: nowInMs,
+    });
+  }, [timeRange.startTime, timeRange.endTime, setTimeRange]);
+
+  useInterval(setTimeRangeToNow, autoRefreshEnabled ? refreshInterval : null);
+
+  const bucketDuration = useMemo(() => {
+    // This function takes the current time range in ms,
+    // works out the bucket interval we'd need to always
+    // display 200 data points, and then takes that new
+    // value and works out the nearest multiple of
+    // 900000 (15 minutes) to it, so that we don't end up with
+    // jaggy bucket boundaries between the ML buckets and our
+    // aggregation buckets.
+    const msRange = timeRange.endTime - timeRange.startTime;
+    const bucketIntervalInMs = msRange / 200;
+    const bucketSpan = 900000; // TODO: Pull this from 'common' when setup hook PR is merged
+    const result = bucketSpan * Math.round(bucketIntervalInMs / bucketSpan);
+    const roundedResult = parseInt(Number(result).toFixed(0), 10);
+    return roundedResult < bucketSpan ? bucketSpan : roundedResult;
+  }, [timeRange]);
+  const { isLoading, logEntryRate } = useLogAnalysisResults({
+    sourceId,
+    startTime: timeRange.startTime,
+    endTime: timeRange.endTime,
+    bucketDuration,
+  });
+  const handleTimeRangeChange = useCallback(
+    ({ start, end }: { start: string; end: string }) => {
+      const parsedStart = dateMath.parse(start);
+      const parsedEnd = dateMath.parse(end);
+      setTimeRange({
+        startTime:
+          !parsedStart || !parsedStart.isValid() ? timeRange.startTime : parsedStart.valueOf(),
+        endTime: !parsedEnd || !parsedEnd.isValid() ? timeRange.endTime : parsedEnd.valueOf(),
+      });
+    },
+    [setTimeRange, timeRange]
+  );
+
+  const anomaliesDetected = useMemo(() => {
+    if (!logEntryRate) {
+      return null;
+    } else {
+      if (logEntryRate.histogramBuckets && logEntryRate.histogramBuckets.length) {
+        return logEntryRate.histogramBuckets.reduce((acc: any, bucket) => {
+          if (bucket.anomalies.length > 0) {
+            return (
+              acc +
+              bucket.anomalies.reduce((anomalyAcc: any, anomaly) => {
+                return anomalyAcc + 1;
+              }, 0)
+            );
+          } else {
+            return acc;
+          }
+        }, 0);
+      } else {
+        return null;
+      }
+    }
+  }, [logEntryRate]);
+
+  return (
+    <>
+      {isLoading && !logEntryRate ? (
+        <>{getLoadingState()}</>
+      ) : (
+        <>
+          <EuiPanel paddingSize="l" grow={false}>
```

Comment:
I agree this looks better - I've switched it over 👍 

> I think it's okay to leave it up there for now, but I was thinking about why we don't move all of this under "Log entries" and that's why, I think (correct me if I'm wrong).

Yep, correct, I was trying to cause the least amount of cognitive WTH when we introduce the other bits. My thinking was the date picker will always be there. And with the text, whilst right now it is only relevant to the anomalies, when the other sections come it will just grow, e.g.: `Detected 11 anomalies, 3 rare messages, and 17 log categories` (or something to that effect!).